Sources/MoonlightTests/MoonlightTestCase.swift

```diff
@@ -37,15 +38,15 @@ public extension MoonlightTestCase {
     ) {
         let completed = expectation(description: String(describing: Self.self))
 
-        events.reduce(Just(initialState).eraseToAnyPublisher()) { newState, newEvent in
-            newState
-                .flatMap { [unowned self] state in
-                    self.transform(state, newEvent, self.environment)
-                        .map { self.apply(state, $0) }
-                }
-                .eraseToAnyPublisher()
-            }
-            .last()
+        Moonlight.start(
+            initialState: initialState,
+            environment: environment,
+            feedback: { _ in events.publisher.eraseToAnyPublisher() },
+            transform: transform,
+            apply: apply,
+            store: { cancellables.update(with: $0) }
+        )
+            .first()
```

Comment:
why first?

Reply:
for cancelling the subscription after getting the result

Reply:
what if we pass it a last that will mutate the state two times, will the second one be ignored?

Reply:
no, it will be the result after all pasted events